enhancements/cloud-integration/tokenized_auth_enablement_operators_on_cloud.md

```diff
@@ -0,0 +1,527 @@
+
+HyperShift changes: Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+OperatorHub changes: Allow for import of additional, well-known, ENV variables on the Subscription object. These fields
+will allows UX for the information needed by CCO or webhook, for input of the cloud credentials. 
+
+OLM changes: Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by 
+parsing the Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended
+before an operator upgrade.
+
+Operator team/ Operator SDK changes: Follow new guidelines for allowing for the operator to work on an STS enabled. 
+New guidelines would include changing CRD to add CredentialRequest references and putting those referenced
+[@jharrington22 need help turning these bullet items into concrete steps]
+- CredentialRequests into a defined directory in the bundle.
+- ServiceAccount annotated:
+- ServiceAccount annotation: Projected ServiceAccount volume but not the Secret
+- Annotate with Role ARN and it projects the Volume
+- Provide CredentialsRequest, but that needs to have ARN added to it.
+- Search for ENV variable (new one) that goes on a Subscription
+- Template it into CredentialsRequest (RoleARN, etc)
+
+Guidance to operator authors to structure controller code to alert on missing cloud credentials, not crash.
+
+SDK to support this new template. SDK to validate and in particular: alert on any permission changes 
+between operator versions.
+
+
+[This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.]
+
+### Workflow Description
+
+[Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.]
+Making a layered operator ready to work on an STS will involve the following steps:
+
+For the operator author team:
+- Add CredentialRequests to the bundle, known location;
+- Add references to the CredentialRequests in the CRD spec;
+- Use SDK to validate bundle to catch permission changes or mis-configuration
```

Comment:
dumb question: would we know this for sure from the CredentialsRequests?

Reply:
A diff of the old and new CredentialsRequest ProviderSpec phrase should be able to tell if there are permission changes.

Mis-configuration, well maybe only gross errors, like claiming STS enablement but not providing any CredentialsRequest(s)?